x/contractmanager/ibc_middleware.go

```diff
@@ -34,9 +36,16 @@ func (k SudoLimitWrapper) Sudo(ctx sdk.Context, contractAddress sdk.AccAddress,
 		// maybe later we'll retrieve actual errors from events
 		resp, err = k.WasmKeeper.Sudo(cacheCtx, contractAddress, msg)
 	}()
-	if err != nil {
-		// the contract either returned an error or panicked with `out of gas`
-		k.contractManager.AddContractFailure(ctx, contractAddress.String(), msg)
+	if err != nil { // the contract either returned an error or panicked with `out of gas`
+		failure := k.contractManager.AddContractFailure(ctx, contractAddress.String(), msg, redactError(err).Error())
```

Comment:
Maybe we should redact an error inside '.AddContractFailure' method?
What if someone forget about non determenism and passes a full error here as an argument?

Reply:
yeah it kind of makes sense and I've even added this as one of the **Left to be done** points. but I got unsure about it just 30 minutes ago while driving home by taxi: it's useful to have such under-the-hood redaction from the from the sudo wrapper's side, but it will be too much for cases when we have a deterministic error passed to the `AddContractFailure` method, e.g. in the InitGenesis func we have concrete errors passing there and they will be all formatted as `codespace: undefined, code: 1` (the default value for errors returned from `redactError`). I think we should not move `redactError` inside the `AddContractFailure` but expect the error to be deterministic on the input, but we definitely should make a giant warning about that somewhere around the `AddContractFailure` method